execution-engine/engine-test-support/src/test_context.rs

```diff
@@ -18,14 +20,83 @@ pub struct TestContext {
 }
 
 impl TestContext {
-    /// Runs the supplied [`Session`], asserting successful execution of the contained deploy and
-    /// subsequent commit of the resulting transforms.
+    fn maybe_purse_balance(&self, purse_uref: Option<URef>) -> Option<Motes> {
+        match purse_uref {
+            None => None,
+            Some(purse_uref) => {
+                let purse_balance = self.get_balance(purse_uref.addr());
+                Some(Motes::new(purse_balance))
+            }
+        }
+    }
+
+    /// Runs the supplied [`Session`], return Result of executing the contained deploy
+    ///
+    /// if Session.expect_success (default) will err if failure.  (Allows cases where failure is
+    /// expected)
+    ///
+    /// if Session.check_transfer_success is given, will verify transfer balances
+    /// including gas used.
+    ///
+    /// if Session.commit (default) will commit resulting transforms.
```

Comment:
The generated docs for this crate are aimed at third party users, so it's worthwhile adding links and not referring to private internals which don't appear in the generated docs.
```suggestion
    /// Runs the supplied [`Session`] checking specified expectations of the execution and
    /// subsequent commit of transforms are met.
    ///
    /// If `session` was built without
    /// [`without_expect_success()`](crate::SessionBuilder::without_expect_success) (the default)
    /// then `run()` will panic if execution of the deploy fails.
    ///
    /// If `session` was built with
    /// [`with_check_transfer_success()`](crate::SessionBuilder::with_check_transfer_success), (not
    /// the default) then `run()` will verify transfer balances including gas used.
    ///
    /// If `session` was built without
    /// [`without_commit()`](crate::SessionBuilder::without_commit) (the default), then `run()` will
    /// commit the resulting transforms.
```